flybywiresim/a32nx
# feat(mcdu): upgrade perf appr page to release 2

~~Partial #1823~~ ~~(still need go-around page, to come in future PR, with changes from #1824)~~ Done in #3650.
Fixes #2096
Fixes #3121

## Summary of Changes

- Pull approach page changes from @beheh's previous PR (#1824)
- Tighten/fix input checks per neo FCOM entry format listing
- Remember pilot entered values and show in large font (automatic/calculated values small)
- Rework approach page appearance to Honeywell release 2 standard
- ~~Allow selection of BARO-RADIO or MDA-DH for minimums~~

## Questions
- Can QNH, Temp, Wind be cleared once entered?
- ~~Does LDG CONF look correct when CONF3 selected? (no references)~~
- ~~Does FLS/FINAL APP look correct when FINAL APP selected? (note FLS is inop at this stage, so greyed out)~~
- ~~What happens when QNH is already entered, and hPa/inHg selection is changed on EFIS?~~

## Screenshots (if necessary)


Empty page, no approach selected:
![appr_empty](https://user-images.githubusercontent.com/9995998/110750589-966ff100-82a7-11eb-89a4-35b4cb62a673.PNG)
Empty page with approach selected:
![appr_empty_radio](https://user-images.githubusercontent.com/9995998/110750650-ac7db180-82a7-11eb-81a9-7cf0988226ec.PNG)
Empty page within 180 nm of destination:
![appr_empty_within180](https://user-images.githubusercontent.com/9995998/110750666-b0a9cf00-82a7-11eb-901b-83cdfe196dc9.PNG)
Filled page with CONF3 selected:
![appr_filled_conf3](https://user-images.githubusercontent.com/9995998/110750682-b6071980-82a7-11eb-8127-ed7d3961e7d3.PNG)
Filled page with non-precision approach selected:
![appr_filled_nonprec](https://user-images.githubusercontent.com/9995998/110750692-bacbcd80-82a7-11eb-926b-738d21eb417d.PNG)
Pilot entered Vapp:
![appr_pilot_vapp](https://user-images.githubusercontent.com/9995998/110750706-bf908180-82a7-11eb-91e5-de902a93c357.PNG)

## References




[A330 with release 2](https://youtu.be/MuP0_u6Q778?t=62):
![image](https://user-images.githubusercontent.com/9995998/108594984-1cec8d80-73e2-11eb-9209-fb1f80108343.png)
FCOM (release 2):
![image](https://user-images.githubusercontent.com/9995998/108595026-5f15cf00-73e2-11eb-99c8-9ccfb3d3ff9e.png)
FCOM (step 1a):
![image](https://user-images.githubusercontent.com/9995998/108595307-1101cb00-73e4-11eb-94d4-fa121908674e.png)
![image](https://user-images.githubusercontent.com/9995998/108595047-7d7bca80-73e2-11eb-941f-114b4e90c6f3.png)
![image](https://user-images.githubusercontent.com/9995998/108595068-a603c480-73e2-11eb-8b5d-2734915dda6d.png)


## Additional context

~~Note an extra column between the left and centre columns used for VLS in this new format. Please consider this for your react MCDU work @Lucky38i.~~ Not required with new font.


Discord username (if different from GitHub): tracernz

## Testing instructions

- Spawn a new flight at the gate and go to PERF APPR page
  - Try various entry formats e.g. 29.92 (in.Hg), 2992 (in.Hg), 1013 (hPa), and variations thereof, including 3 digit hPa, also illegal ones you can think of, and make sure you can't accidentally enter a temperature
  - Switch LDG FLAPs config and observe the lines VLS and VAPP updating appropriately
  - Override VAPP and ensure the font goes large for pilot entry, then clear to revert to automatic calculation
  - Repeat with TRANS ALT
  - Enter various winds, valid and invalid to ensure correct display, with leading zeroes.
- Spawn a flight a little over 180 Nm from destination, enter flightplan but don't enter APPR values.
  - Check that QNH and TEMP turn to amber boxes when 180 Nm out
  - Select ILS approach on F-PLN and ensure both BARO and RADIO entries are available on APPR, FINAL APP/FLS are hidden, and page title shows selected approach appropriately
  - Select non-precision approaches (RNAV, VOR-DME, etc.), check that only BARO entry is available, and page title shows selected approach appropriately
- Fly an approach or two and make sure you're happy!


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
